core/src/main/java/com/sequenceiq/cloudbreak/service/cost/ClusterCostService.java

```diff
@@ -0,0 +1,80 @@
+package com.sequenceiq.cloudbreak.service.cost;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.common.cost.RealTimeCost;
+import com.sequenceiq.cloudbreak.cost.co2.CarbonCalculatorService;
+import com.sequenceiq.cloudbreak.cost.model.ClusterCostDto;
+import com.sequenceiq.cloudbreak.cost.usd.UsdCalculatorService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+@Service
+public class ClusterCostService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterCostService.class);
+
+    @Inject
+    private CarbonCalculatorService carbonCalculatorService;
+
+    @Inject
+    private UsdCalculatorService usdCalculatorService;
+
+    @Inject
+    private InstanceTypeCollectorService instanceTypeCollectorService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private StackService stackService;
+
+    public Map<String, RealTimeCost> getCosts(List<String> crns) {
+        Map<String, RealTimeCost> realTimeCosts = new HashMap<>();
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        boolean usdCalculationEnabled = entitlementService.isUsdCostCalculationEnabled(accountId);
+        boolean co2CalculationEnabled = entitlementService.isCO2CalculationEnabled(accountId);
+
+        if (usdCalculationEnabled || co2CalculationEnabled) {
```

Comment:
At this point at least one of them is probably true otherwise you exit with no entitlement enabled, so need for this check here.